### PR TITLE
enhancement: make assignment of network_adapters optional

### LINKED ISSUE #202: Make network adapters optional when creating a new VM
#### Community Note

Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request.
If you are interested in working on this issue or have submitted a pull request, please leave a comment.

#### Description

Currently, when trying to create a VM without specifying a network adapter the build will fail.
The network_adapters option is not optional although it is not necessary for a complete build.

#### Use Case(s)

When you don't want to expose the VM to a network (security concerns).

### PR DESCRIPTION
**Pull request by cibofo**
Removes the check that prevents the creation of a VM without a network adapter, thus allowing to create a VM without network adapters 

Closes #202

### CODE REVIEW

**Comment by tenthirtyam** [comment]
Recommendations:

**Title**: enhancement: make assignment of `network_adapters` optional

**Labels**: `enhancement`

**Docs**: Update the in-code docs for **Network Configuration** to denote that `network_adapters` is optional but with any caveats and limitations. Then generate the docs with `packer-sdc`.

cc @nywilken 

**Ryan Johnson**
Staff II Solutions Architect | VMware, Inc.

File: builder/vsphere/iso/step_create.go
Comment (tenthirtyam) [inline builder/vsphere/iso/step_create.go:72]: "// Sets" with the `[SPACE]`

**Comment by cibofo** [comment]
> Were these manual edits? This is derived from the code and generated with packer-sdc.
> 
No, it was using  `go generate ./builder/vsphere/iso/step_create.go`

File: docs-partials/builder/vsphere/iso/CreateConfig-not-required.mdx
Comment (tenthirtyam) [inline docs-partials/builder/vsphere/iso/CreateConfig-not-required.mdx:12]: This update would be generated with `packer-sdc generate`.

Comment (cibofo) [reply docs-partials/builder/vsphere/iso/CreateConfig-not-required.mdx:12]: I don't think I know what you mean, there is not `generate` command for packer-sdc

File: builder/vsphere/iso/step_create.go
Comment (tenthirtyam) [reply builder/vsphere/iso/step_create.go:72]:
```go
// Sets the network adapters, if no adapter is specified, every network-related task is not applicable.
  ^ [SPACE]
```

**Comment by cibofo** [comment]
Any updates?

**Comment by tenthirtyam** [comment]
@cibofo Any update for what in particular?

**Comment by cibofo** [comment]
> @cibofo Any update for what in particular?

On the merge, is there something I should do?

**Comment by tenthirtyam** [comment]
@cibofo it's pending review by the maintainers.

Refer to the Pull Request Lifecycle in the [CONTRIBUTING.md](https://github.com/hashicorp/packer-plugin-vsphere/blob/main/.github/CONTRIBUTING.md).

**Ryan Johnson**
Staff II Solutions Architect | VMware, Inc.

**Comment by cibofo** [comment]
Oh, thanks

**Review by nywilken — APPROVED** [review]
The changes here look good. Thanks for updating the documentation.

**Pull request merged**
